### PR TITLE
[BugFix] Alter table with changing buckets parse fix

### LINKED ISSUE #44524: [BUG] Alter table occur parse exception
Alter table had bug for optimze buckets.

### Steps to reproduce the behavior (Required)
1. CREATE TABLE `ods_datalake_sr` (
  `pk_names` varchar(65533) NOT NULL COMMENT "主键",
  `operation_ts` bigint(20) NULL COMMENT "比较键",
  `operation_type` varchar(10) NULL COMMENT "binlog数据类型",
  `db_name` varchar(20) NULL COMMENT "库名",
  `tb_name` varchar(100) NULL COMMENT "表名",
  `id` int(11) NULL COMMENT "id主键",
  `user_id` varchar(65533) NULL COMMENT "id",
  `name` varchar(65533) NULL COMMENT "name",
  `from` varchar(65533) NULL COMMENT "来源",
  `model` varchar(65533) NULL COMMENT "号"
) ENGINE=OLAP 
PRIMARY KEY(`pk_names`)
DISTRIBUTED BY HASH(`pk_names`) BUCKETS 90 
PROPERTIES (
"replication_num" = "2",
"in_memory" = "false",
"enable_persistent_index" = "true",
"replicated_storage" = "true",
"compression" = "LZ4"
); 

**had a col named `from` in table**

2.ALTER TABLE ods_datalake_srDISTRIBUTED BY HASH(pk_names) BUCKETS 75;  

3. show alter table OPTIMIZE ;  
ALTER DISTRIBUTED BY HASH(pk_names) BUCKETS 75 | 1739880       | CANCELLED | replace partitions failed com.starrocks.alter.AlterCancelException: all partitions rewrite failed                                  | 99       | 86400   |

4. fe log:
2024-04-22 14:38:43,344 WARN (starrocks-taskrun-pool-211|3616532) [TaskRunExecutor.lambda$executeTaskRun$0():57] failed to execute TaskRun.
com.starrocks.sql.parser.ParsingException: Getting syntax error at line 1, column 216. Detail message: Unexpected input 'from', the most similar input is {a legal identifier}.
 at com.starrocks.sql.parser.ErrorHandler.syntaxError(ErrorHandler.java:40) ~[starrocks-fe.jar:?]
 at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41) ~[antlr4-runtime-4.9.2.jar:4.9.2]
 at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544) ~[antlr4-runtime-4.9.2.jar:4.9.2]
 at com.starrocks.sql.parser.SqlParser$1.reportInputMismatch(SqlParser.java:222) ~[starrocks-fe.jar:?]
 at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139) ~[antlr4-runtime-4.9.2.jar:4.9.2]
 at com.starrocks.sql.parser.StarRocksParser.selectItem(StarRocksParser.java:35224) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.querySpecification(StarRocksParser.java:34229) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.queryPrimary(StarRocksParser.java:33729) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.queryNoWith(StarRocksParser.java:33435) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.queryRelation(StarRocksParser.java:33312) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.queryStatement(StarRocksParser.java:33244) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.insertStatement(StarRocksParser.java:16019) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.statement(StarRocksParser.java:1852) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.singleStatement(StarRocksParser.java:756) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.StarRocksParser.sqlStatements(StarRocksParser.java:647) ~[starrocks-fe.jar:?]
 at com.starrocks.sql.parser.SqlParser.parseWithStarRocksDialect(SqlParser.java:103)

### Expected behavior (Required)
success alter table with buckets change

### Real behavior (Required)
error out for parse error

### StarRocks version (Required)
3.2.4

### PR DESCRIPTION
**Pull request by xuzifu666**
## What I'm doing:

Fixes https://github.com/StarRocks/starrocks/issues/44524

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by meegoo** [comment]
Thanks for your PR, please add a sql test to cover this case @xuzifu666

**Comment by xuzifu666** [comment]
> Thanks for your PR, please add a sql test to cover this case @xuzifu666

OK，would add sqltest latter.

File: test/sql/test_alter_table/R/test_alter_table_abnormal [hunk 1]
```diff
@@ -281,4 +281,19 @@ ALTER TABLE site_access1 DROP COLUMN site_id;
 -- result:
 E: (1064, 'Distribution column[site_id] cannot be dropped. index[site_access1]')
 -- !result
+-- name: test_alter_key_buckets
```
Comment (meegoo) [inline test/sql/test_alter_table/R/test_alter_table_abnormal]: need to delete

Comment (xuzifu666) [reply test/sql/test_alter_table/R/test_alter_table_abnormal]: Had changed to optimize table，Thanks~ @meegoo

**Review by meegoo — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/alter/OptimizeJobV2.java [hunk 2]
```diff
@@ -259,7 +259,7 @@ protected void runWaitingTxnJob() throws AlterCancelException {
                         }
             );
             tableCoumnNames = targetTable.getBaseSchema().stream().filter(column -> !column.isGeneratedColumn())
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/alter/OptimizeJobV2.java:261]: please also helps modify this typo.

Comment (xuzifu666) [reply fe/fe-core/src/main/java/com/starrocks/alter/OptimizeJobV2.java]: Done @packy92

**Review by packy92 — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Pull request merged**